GREEDY/ImagePreparation/ImageFormating.cs

```diff
@@ -1,171 +1,117 @@
-                new IntPoint(Convert.ToInt32(cornerPoints[2]), Convert.ToInt32(cornerPoints[3])),
-                new IntPoint(Convert.ToInt32(cornerPoints[4]), Convert.ToInt32(cornerPoints[5])),
-                new IntPoint(Convert.ToInt32(cornerPoints[6]), Convert.ToInt32(cornerPoints[7]))
-            };
-            QuadrilateralTransformation filter = new QuadrilateralTransformation(corners);
-            Bitmap newImage = filter.Apply(bitmap);
-            return newImage;
+            Bitmap edited;
+            edited = Binarization(bitmap);
+            edited = RemoveNoise(edited);
+            edited = BiggestBlob(edited);
+            edited = Rotate(edited);
+            edited = Deskew(edited);
+            edited = BiggestBlob(edited);
+            edited = Rescale(edited);
+            return edited;
         }
 
-        //returns the bigest found rectangle
-        private Bitmap FilterCropedImages(List<Bitmap> list)
+        //Not really sure what it does but the recomendations suggest it helps the OCR read better
+        private Bitmap Rescale(Bitmap bitmap)
         {
-            if (list.Any())
-            {
-                // this guy throws an exception if list is null
-                return list.MaxBy(x => x.Height * x.Width);
-            }
-            else
-            {
-                return null;
-            }
+            Bitmap rescaledBitmap = new Bitmap(bitmap);
+            rescaledBitmap.SetResolution(300, 300); //recomended dpi for OCR
+            return rescaledBitmap;
         }
 
-        public Bitmap FormatImage(Bitmap bitmap)
+        /// <summary>
+        /// Removes noise (small dots/smudges from an image)
+        /// </summary>
+        /// <param name="bitmap"></param>
+        /// <returns></returns>
+        private Bitmap RemoveNoise(Bitmap bitmap)
         {
-            if (bitmap.Width > bitmap.Height)
-            {
-                bitmap.RotateFlip(RotateFlipType.Rotate90FlipNone);
-            }
-                
-            int originalWidth = bitmap.Width;
-            int originalHeight = bitmap.Height;
-
-            //resizing is needed for better rectangle detection
-            Image < Bgr, Byte > img = new Image<Bgr, byte>(bitmap).Resize(400, 400, Inter.Linear, true);
-
-            int resizedWidth = img.Width;
-            int resizedHeight = img.Height;
+            Image<Gray, byte> image = new Image<Gray, byte>(bitmap);
 
-            //Convert the image to grayscale and filter out the noise
-            UMat uimage = new UMat();
-            CvInvoke.CvtColor(img, uimage, ColorConversion.Bgr2Gray);
+            //Image<Gray, byte> edited = image.SmoothBlur(10, 10, true);
+            Image<Gray, byte> edited = image.SmoothMedian(7);//(15);
+            //Image<Gray, byte> edited = image.SmoothBilatral(7, 255, 34);
+            //Image<Gray, byte> edited = image.SmoothGaussian(3, 3, 34.3, 45.3);
```

Comment:
What are these commented lines?

Reply:
yes, you should remove during testing created parts